src/libexec/zmrestoreblob

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/perl
+#
+# ***** BEGIN LICENSE BLOCK *****
+# Zimbra Collaboration Suite Server
+# Copyright (C) 2009, 2018 Synacor, Inc.
+# This program is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software Foundation,
+# version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along with this program.
+# If not, see <https://www.gnu.org/licenses/>.
+# ***** END LICENSE BLOCK *****
+
+#Usage 
+# su - zimbra
+# zmrestoreblob -m 3 
+# where -m is to specify which mailbox to check. In above case it is mailbox id 3 
+
+use strict;
+use warnings;  #or else.
+use lib "/opt/zimbra/common/lib/perl5";
+use Data::Dumper;
+use File::Basename;
+use File::Copy;
+
+my $matched_blob;
+
+my $backup_dir = "/opt/zimbra/backup/blobs";
+
+my $mailbox_id = $ARGV[1];
+
+#execute zmblobchk command for a given mailbox.
+my $zmblobchk_output = `zmblobchk -m $mailbox_id start`;
+
+# check if there are any orphaned headers.
+my $orphaned_header_check = ($zmblobchk_output =~ /blob not found/g);
+
+chop($zmblobchk_output);
+
+if ($orphaned_header_check) {
+
+	# find the missing blob entries
+	my @matches = ($zmblobchk_output =~ /(\/opt\/zimbra\/store.*\.msg)/gm);
+
+	for $matched_blob(@matches) {		
```

Comment:
@ajinkya81 Probably want to run your script through [PearlCritic](http://perlcritic.com/).  I just pasted it in and ran a quick check. At the most permissive level of checking (least warnings). It complained about this line.

    Loop iterator is not lexical at line 48, column 2. See page 108 of PBP.

Also, you might want to add some reviewers to your pull request. :-)

Reply:
fixed